File: OsmAndWear/src/net/osmand/plus/api/SQLiteAPI.java
package net.osmand.plus.api;


import androidx.annotation.Nullable;

public interface SQLiteAPI {

	interface SQLiteConnection {
		
		void close();
		
		SQLiteCursor rawQuery(String sql, String[] selectionArgs);

		void execSQL(String query);
		
		void execSQL(String query, Object[] objects);

		SQLiteStatement compileStatement(String string);

		void setVersion(int newVersion);
		
		int getVersion();

		boolean isReadOnly();

		boolean isClosed();

	}
	
	interface SQLiteCursor {
		
		String[] getColumnNames();

		boolean moveToFirst();
		
		boolean moveToNext();

		/**
		 * Takes parameter value (zero based)
		 */
		String getString(int ind);

		double getDouble(int ind);
		
		long getLong(int ind);
		
		int getInt(int ind);
		
		byte[] getBlob(int ind);
		
		boolean isNull(int ind);

		int getColumnIndex(String columnName);
		
		void close();

	}
	
	interface SQLiteStatement {

		// 1 based argument
		void bindString(int i, String filterId);

		// 1 based argument
		void bindNull(int i);

		void execute();

		void close();

		long simpleQueryForLong();

		String simpleQueryForString();

		void bindLong(int i, long val);
		
		void bindBlob(int i, byte[] val);

	}

	@Nullable
	SQLiteConnection getOrCreateDatabase(String name, boolean readOnly);

	@Nullable
	SQLiteConnection openByAbsolutePath(String path, boolean readOnly);
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/api/SQLiteAPI.kt
package net.osmand.shared.api

interface SQLiteAPI {

	interface SQLiteConnection {
		fun close()
		fun rawQuery(sql: String, selectionArgs: Array<String>?): SQLiteCursor?
		fun execSQL(query: String)
		fun execSQL(query: String, objects: Array<Any?>)
		fun compileStatement(query: String): SQLiteStatement
		fun setVersion(newVersion: Int)
		fun getVersion(): Int
		fun isReadOnly(): Boolean
		fun isClosed(): Boolean
	}

	interface SQLiteCursor {
		fun getColumnNames(): Array<String>
		fun moveToFirst(): Boolean
		fun moveToNext(): Boolean
		fun getString(ind: Int): String
		fun getDouble(ind: Int): Double
		fun getLong(ind: Int): Long
		fun getInt(ind: Int): Int
		fun getBlob(ind: Int): ByteArray
		fun isNull(ind: Int): Boolean
		fun getColumnIndex(columnName: String): Int
		fun close()
	}

	interface SQLiteStatement {
		fun bindString(i: Int, value: String)
		fun bindNull(i: Int)
		fun execute()
		fun close()
		fun simpleQueryForLong(): Long
		fun simpleQueryForString(): String
		fun bindLong(i: Int, value: Long)
		fun bindBlob(i: Int, value: ByteArray)
	}

	fun getOrCreateDatabase(name: String, readOnly: Boolean): SQLiteConnection?
	fun openByAbsolutePath(path: String, readOnly: Boolean): SQLiteConnection?
}
